include/multipass/url_downloader.h

```diff
@@ -43,13 +40,15 @@ class URLDownloader
                              const ProgressMonitor& monitor);
     virtual QByteArray download(const QUrl& url);
     virtual QDateTime last_modified(const QUrl& url);
+    virtual void abort_all_downloads();
 
 private:
     URLDownloader(const URLDownloader&) = delete;
     URLDownloader& operator=(const URLDownloader&) = delete;
 
     const Path cache_dir_path;
     std::chrono::milliseconds timeout;
+    bool abort_download{false};
```

Comment:
`abort_download` is read and written to concurrently, so strictly speaking it requires synchronization. 

I have used the same technique as you a few times before because, in practice, since it is a boolean, the worse that I could see happening is `on_download` not seeing it as true until a later iteration. Even it the underlying memory got messed up &ndash; I thought &ndash; it would still resolve to `true`. However, this assumes a memory model, and forgets the messing up could actually mean setting it to 0 (see below). Formally it is undefined behavior, so, better keep it sane.

A second issue is that, if the compiler caches it in a register, the second thread might not see it changing ever (until there is a sync with memory, which may never happen IIUC). This can be thought of as a particular case of the memory being messed up &ndash; to the previous value, in this case. That could be prevented by declaring it `volatile`, but that does nothing for our undefined behavior yet...

So, thr right app approach I think is to declare it `std::atomic_bool`. I believe at that point it can't be cached, so no point declaring it volatile.

Note for completeness: `std::atomic` (which `atomic_bool` specializes) uses a lock underneath, so there is a little performance penalty. It does not matter in our case but, if it did, there would be another alternative: `std::atomic_flag`. Unlike the former, which is an `std::atomic` specialization, the latter is guaranteed to be lock-free. Its interface is awkward though.

Reply:
@ricab, hey, thanks for the review.  I had thought about using an atomic, but thinking that we would only ever have one writer (and potentially many readers), that it would be safe.  But after reading what you wrote and doing a bit of research, yeah, atomic is the way to go.  I'll change that.

Reply:
Done